install/operator/.lib.sh

```diff
@@ -39,71 +39,51 @@ readopt() {
     echo $default
 }
 
-
-docker_is_available() {
-    if [[ "$(which docker)" == "" ]] ; then
-        echo "warn: docker command not installed"
-        echo
-        return 0
-    fi
-
-    if docker info > /dev/null 2> /dev/null ; then
-        return 1
-    else
-        echo "warn: the docker comamand is not conected to a server"
-        echo
-        return 0
-    fi
-}
-
 build_operator()
 {
     local strategy="$1"
     shift
-    local gopackage="$1"
-    shift
 
-    if [ "$strategy" == "auto" ] ; then
-
-        if docker_is_available; then
-            if [[ "$(which go)" == "" ]] ; then
-                echo 'error: you must have either docker or go installed to build this project.'
-                exit 1
-            else
-                strategy="go"
-            fi
+    local hasgo=$(go_is_available)
+    local hasdocker=$(docker_is_available)
+
+    if [ "$strategy" = "auto" ] ; then
+        if [ "$hasgo" = "OK" ]; then
+            # Prefer go over docker - TODO: do we prefer this?
+            strategy="go"
+        elif [ "$hasdocker" = "OK" ]; then
+            # go not available so try docker
+            printf "WARN: Building with 'docker' since 'go' command is not available ... \n\t\t$hasgo\n"
+            strategy="docker"
         else
-            if [[ "$(which go)" == "" ]] ; then
-                echo 'warn: building with docker since you do not have go installed.'
-                strategy="docker"
-            else
-                strategy="go"
-            fi
+            echo "ERROR: Building the operator requires either 'docker' or 'go' commands to be installed."
+            exit 1
         fi
     fi
 
-
     case "$strategy" in
     "go")
+        if [ "$hasgo" != "OK" ]; then
+            echo "$hasgo"
+            exit 1
+        fi
+
         echo ======================================================
         echo Building executable with go tooling
         echo ======================================================
         export GO111MODULE=on
 
         go mod vendor
-        if [[ "$(which operator-sdk)" != "" ]] ; then
-            if [[ "$(pwd)" != "$GOPATH/src/${gopackage}" ]] ; then
-                echo
-                echo "warnning: operator-sdk only works on project's in the \$GOPATH"
-                echo "          can't use it to update the generated code"
-                echo "          please move this project under the \$GOPATH so that :'$(pwd)'"
-                echo "          is located at '$GOPATH/src/${gopackage}'"
-                echo
-            else
-                operator-sdk generate k8s
-                # operator-sdk generate openapi
-            fi
+
+        local hassdk=$(operatorsdk_is_available)
+        if [ "$hassdk" = "OK" ]; then
```

Comment:
```suggestion
        if [ "$hassdk" == "OK" ]; then
```

Reply:
Didn't the scripts need to be shell agnostic and avoid specific implementations?

Reply:
Even if that was the case, which I'm fairly certain it isn't, that ship has sailed, we're very bash-specific in our shell scripts.

Reply:
Was just something Roland commented on in one of my first PRs late last year.

Reply:
I'm sure he would want us to be better at that, I'm not sure if that's even possible, I think we're using quite a lot of bashisms at this point; also not sure if that's required bash is pretty much universal `sh`, the only use case would be within a busybox based distribution, and I don't think we have that use case.